### DISCUSSION TITLE
auto hide history / logbook when not recorded

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [NemesisRE/kiosk-mode#155] [Ideas]
<img width="587" alt="Scherm­afbeelding 2023-12-15 om 16 15 21" src="https://github.com/NemesisRE/kiosk-mode/assets/33354141/e91537e3-55ce-40e5-bf1a-e3ac978904b3">

It has always bugged me Home Assistant keeps showing these 'No history found/No logbook entries found/ messages on the more-info panels, when it knows these entities are not recorded. Moreover, it shows the link to the respective views for logbook and history, only to find out there, or be reconfirmed, these entities are not recorded.....

Aware there probably is a wall between the backend config and frontend, but why shouldnt we be able to copy the final included entities in some more-info-display configuration and have Kiosk-mode 'filter' those from showing history and logbook automatically.

I have it ready for c&p. 

<img width="339" alt="Scherm­afbeelding 2023-12-15 om 16 24 03" src="https://github.com/NemesisRE/kiosk-mode/assets/33354141/89a5817c-84ff-443a-a0d3-22dcd964cee1">

Or even an include could be possible
```
kiosk_mode:
  history_logbook: !include ../../included_entities
```
just like I do with recorder:

```
# db_url: !secret db_url
purge_keep_days: 7

# https://www.home-assistant.io/integrations/recorder/#configure-filter
exclude:
  !include excluded_entities.yaml

include:
  !include included_entities.yaml
```

Seems your plugin is the ultimate tool to fix that oversight.

Ofc you'd be most welcome to add it to custom_attributes too, but then we'd need to rename it to custom-more-info ;-)

anyways, hope this makes sense

**Comment by elchininet** [comment]
@Mariusthvdb,

That kind of feature falls on the side of a personal preference or convenience. Kiosk mode is not intended to satisfy what is convenient to show or not, but an specific and focused task: *avoid unauthorized users interact with certain critical elements that are being presented in a kiosk device*. If you don‘t want your users see the history or the logbook, you can hide them with `kiosk-mode`. Choosing if you want to hide them or not depending if they look nice or not, is not in the scope of the plugin.

But of course @NemesisRE should have the last call on this.

>Ofc you'd be most welcome to add it to custom_attributes too, but then we'd need to rename it to custom-more-info ;-)

I would not add it there either but it is up to you, that plugin that you created has a name and a different intention. If you want to do more stuff, as you said, you would need to change its name and scope, but if you follow that path you will be changing its name and scope again when something new comes to your mind. So my personal recommendation, create a separated plugin for that. Plugins should have a specific and clear task to do, otherwise it is a never ending and you will end-up very quickly in something very hard to maintain and escalate.

But that is just my recomendation, adding stuff there is up to you 🙂

**Reply by Mariusthvdb** [reply]
yeah, I guess this falls somewhere in the middle.
its not because its not looking nice or not. Nor some frivolous personal liking. It's a fluke in HA design imho, showing history and logbook placeholders for items that have no history or logbook. As they very well know, because its explicitly stated in the Statistics page for these entities

Moreover, It leads the user to a (not necessarily unauthorized but) misleading Show More page, where again, there is nothing to view.

I dare to say its a design bug even because of that.

So, appreciate your philosophy, and ys, maybe a separate plugin could do it. Though Id love to minimize the number of plugins we need to get a better Frontend...... o well, good night for now ;-)

**Reply by NemesisRE** [reply]
> But of course @NemesisRE should have the last call on this.


Exactly what you said.

Plugins should always have one clearly defined purpose and if something falls out of this defined scope you should have another plugin.

I agree with you that this behavior could be a design flaw so why don't you try to discuss this in the ha repo? If you post a link I would support it.

**Reply by Mariusthvdb** [reply]
haha I have long time ago https://github.com/home-assistant/frontend/discussions/10277

maybe we should revive that... let me see if I can c&p some of the info I posted above

**Comment by elchininet** [comment]
Closing this discussion, if someone reaches here trying to auto hide history and logbook, @Mariusthvdb created a HACS plugin for that: [custom-more-info](https://github.com/Mariusthvdb/custom-more-info).
